crates/wallet/src/wallet/mod.rs

```diff
@@ -2312,6 +2303,15 @@ impl Wallet {
         }
     }
 
+    /// Get a mutable reference of the staged [`ChangeSet`] that is yet to be commited (if any).
+    pub fn staged_mut(&mut self) -> Option<&mut ChangeSet> {
```

Comment:
I was thinking that this method may be useful for some users. I.e. they don't need to query the staged area twice if they wanted to do something like we do in `PersistedWallet::persist`.

Reply:
For the sake of discussion I think you can do it without ever getting the stage mutably such as having a method on `Wallet` that resets the stage but doesn't return it, and this would prevent users from freely modifying their changesets.

```rust
  /// Clears the stage.
  pub fn clear_stage(&mut self) { self.stage = ChangeSet::default(); }
```

Then `persist` would look like

```rust
  /// Persist staged changes.
  pub fn persist(&mut self, persister: &mut P) -> Result<bool, P::Error> {
      if let Some(change) = self.inner.staged() {
          P::persist(persister, change)?;
          self.inner.clear_stage();
          return Ok(true);
      }
      Ok(false)
  }
```